pypdf/_xobj_image_helpers.py

```diff
@@ -148,9 +148,12 @@ def _extended_image_frombytes(
         img = Image.frombytes(mode, size, data)
     except ValueError as exc:
         nb_pix = size[0] * size[1]
-        if len(data) % nb_pix != 0:
+        data_length = len(data)
+        if data_length == 0:
+            raise ValueError("Data is 0 bytes, cannot process an image from empty data.") from exc
```

Comment:
We probably want to use one subclass of `PyPdfError` here (https://github.com/py-pdf/pypdf/blob/main/pypdf/errors.py) - either a new one or an appropriate of the existing - to indicate that we are aware of this and cannot do much about this.

Reply:
Great idea! I've added the new `EmptyImageDataError` to handle cases like these in my latest commit.